web/src/stores/dandiset.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-use-before-define */
+
+import { defineStore } from 'pinia';
+
+import axios from 'axios';
+import RefParser from '@apidevtools/json-schema-ref-parser';
+
+// eslint-disable-next-line import/no-cycle
+import { dandiRest } from '@/rest';
+import { User, Version } from '@/types';
+import { draftVersion } from '@/utils/constants';
+
+interface State {
+  dandiset: Version | null;
+  versions: Version[] | null,
+  loading: boolean,
+  owners: User[] | null,
+  schema: any,
+}
+
+export const useDandisetStore = defineStore('dandiset', {
+  state: (): State => ({
+    dandiset: null,
+    versions: null,
+    loading: false, // No mutation, as we don't want this mutated by the user
+    owners: null,
+    schema: null,
+  }),
+  getters: {
+    version: (state) => (state.dandiset ? state.dandiset.version : draftVersion),
```

Comment:
Can we apply TS types to these functions? It would make the list of getters that much more legible.

Reply:
Unlike Vuex with `direct-vuex`, Pinia is fully typed, so the types of every function parameter/return value here are inferred automatically without needing to manually specify them.

Reply:
Oh, maybe I'm misunderstanding what you're saying. Are you just asking for explicit types to be added? Like
```suggestion
    version: (state): string => (state.dandiset ? state.dandiset.version : draftVersion),
```

Reply:
Yeah, I meant the latter. It's not necessary to fill them all in now, but I think one of the major benefits of TypeScript is that the types are checked by the machine, but also legible to the human.